kcapp-smartboard.js

```diff
@@ -102,7 +111,67 @@ function connectToMatch(data) {
             } else {
                 debug("Already connected to board...");
                 leg.emit('announce', { type: 'success', message: 'Already Connected' });
-                leg.on('leg_finished', disconnectListener.bind(this));
+                smartboard.initialize(peripheral, config.smartboard_button_number,
```

Comment:
Does the GranBoard really require us to reconnect here? 
For the Unicorn board, I believe this won't work, since we are already connected.

If it does we should check here for which board currently is in use, and then only `initialize` if it's a GranBoard, otherwise skip.

It also seems like this is exactly the same as lines 48:87, so we could also extract those out into a separate method to reduce code duplication

Reply:
Yes, the majority of the code of the first branch is duplicated here.

Reconnection does not seem necessary, but the listener is also in the initialize function and thus it was the easiest way of establishing this. 

It might work with less code, but except for one error thrown because we are already connected the code just works.

It also allows to keep kcapp-smartboard running as a daemon which will connect to new legs and to the board if you switch it on again. No need to launch it for each session.

Reply:
So you need to run the initialize function for each leg on the Granboard? Does it also keep track of legs played and stuff like that and needs a reset?
For the `Unicorn Smartboard` it just emits whatever you throw, so it doesn´t need to reestablish it. I also don´t currently have easy access to a Unicorn board to test this change, but I'm a bit worried that calling initialize again here might cause issues, since we are already connected and established above.

Regarding reconnecting, this was solved by having the "Reconnect Smartboard" button in the Frontend

* From a Leg, click "Options" => "Change Order", this will bring up the following dialog
![image](https://github.com/user-attachments/assets/6a74f7f2-04ed-41e5-94cd-ff8c322f2ec4)
Here you can click "Reconnect Smartboard" and it should make the request to reconnect the board if it was restarted/disconnected somehow.

This button will only be available if a `Venue` is selected when starting the match with `smartboard = true`, so make sure you configure your venues correctly by going to `<kcapp>/offices` and adding/editing a venue and enabling smartboard
![image](https://github.com/user-attachments/assets/5a2eb66b-ca5b-418d-9c10-ffe48245d3ff)

Could you see if this works for your case as well?